src/lib/Nav.svelte

```diff
@@ -23,22 +24,40 @@
     }
   }
 
+  function handleScroll() {
+    if (isMenuOpen) {
+      isMenuOpen = false;
+    }
+  }
+
+  function handleClickOutside(event) {
+    const dropdown = document.querySelector('.dropdown-menu');
+    if (isMenuOpen && dropdown && !dropdown.contains(event.target)) {
+      isMenuOpen = false;
+    }
+  }
+
   onMount(() => {
-    document.addEventListener('click', closeResources);
+    window.addEventListener('scroll', handleScroll);
+    document.addEventListener('click', handleClickOutside);
+
     return () => {
-      document.removeEventListener('click', closeResources);
+      window.removeEventListener('scroll', handleScroll);
+      document.removeEventListener('click', handleClickOutside);
     };
   });
```

Comment:
This looks like a use for svelte actions:
https://svelte.dev/tutorial/svelte/actions

what's happening here?
trying to close an overlay by clicking outside the element?